bin/php-language-server.php

```diff
@@ -10,6 +12,21 @@
     }
 }
 
-$server = new LanguageServer(new ProtocolStreamReader(STDIN), new ProtocolStreamWriter(STDOUT));
+if (count($argv) >= 3 && $argv[1] == '--tcp') {
+    $address = $argv[2];
+    $socket = stream_socket_client('tcp://' . $address, $errno, $errstr);
```

Comment:
Only the listen socket is closed, the established connection keeps going. It just won't receive any additional connections after the first one.